### PR TITLE
chore: move merge logic to CTP

### PR DESCRIPTION
**Pull request by zachaller**
fixes: https://github.com/argoproj-labs/gitops-promoter/issues/73

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -201,7 +201,7 @@ ENVTEST_VERSION ?= release-0.19
 GOLANGCI_LINT_VERSION ?= v1.54.2
 MOCKERY_VERSION ?= v2.42.2
 NILAWAY_VERSION ?= latest
-GINKGO_VERSION ?= latest
+GINKGO_VERSION ?= v2.21.0
```
Comment (crenshaw-dev) [inline Makefile]: Probably should set up Renovate for this stuff.

File: api/v1alpha1/changetransferpolicy_types.go [hunk 2]
```diff
@@ -40,6 +40,10 @@ type ChangeTransferPolicySpec struct {
 	// +kubebuilder:validation:Required
 	ActiveBranch string `json:"activeBranch"`
 
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	AutoMerge *bool `json:"autoMerge,omitempty"`
```
Comment (crenshaw-dev) [inline api/v1alpha1/changetransferpolicy_types.go:45]: Need to update example CRDs to reflect this.

File: api/v1alpha1/constants.go [hunk 3]
```diff
@@ -7,8 +7,8 @@ const CommitStatusLabel = "promoter.argoproj.io/commit-status"
 // CommitStatusCopyLabel is the label used to identify copied commit statuses (true or false)
 const CommitStatusCopyLabel = "promoter.argoproj.io/commit-status-copy"
 
-// CopiedProposedCommitPrefixNameLabel is the prefix name for copied proposed commits
-const CopiedProposedCommitPrefixNameLabel = "proposed-"
+// PreviousEnvProposedCommitPrefixNameLabel is the prefix name for copied proposed commits
```
Comment (crenshaw-dev) [inline api/v1alpha1/constants.go:10]: It's hard to understand what "copied proposed commits" means in this context.

File: Makefile (hunk 1 above)
Comment (zachaller) [reply Makefile]: Yea this particular case could be set to pin the same version in gomod that is why it needed to be set

File: internal/controller/changetransferpolicy_controller.go [hunk 4]
```diff
@@ -398,3 +382,67 @@ func (r *ChangeTransferPolicyReconciler) creatOrUpdatePullRequest(ctx context.Co
...
+	logger := log.FromContext(ctx)
+
+	for i, status := range ctp.Status.Proposed.CommitStatuses {
+		if status.Phase != string(promoterv1alpha1.CommitPhaseSuccess) {
+			logger.Info("Proposed commit status is not success", "key", ctp.Spec.ProposedCommitStatuses[i].Key)
+			return nil
+		}
+	}
+
+	if *ctp.Spec.AutoMerge {
```
Comment (crenshaw-dev) [inline internal/controller/changetransferpolicy_controller.go:397]: Need a nil check?

File: internal/controller/changetransferpolicy_controller.go [hunk 5]
```diff
@@ -398,3 +382,67 @@ func (r *ChangeTransferPolicyReconciler) creatOrUpdatePullRequest(ctx context.Co
...
 	return nil
 }
+
+// mergePullRequests tries to merge the pull request if all the checks have passed and the environment is set to auto merge.
+func (r *ChangeTransferPolicyReconciler) mergePullRequests(ctx context.Context, ctp *promoterv1alpha1.ChangeTransferPolicy) error {
+	logger := log.FromContext(ctx)
+
+	for i, status := range ctp.Status.Proposed.CommitStatuses {
+		if status.Phase != string(promoterv1alpha1.CommitPhaseSuccess) {
+			logger.Info("Proposed commit status is not success", "key", ctp.Spec.ProposedCommitStatuses[i].Key)
```
Comment (crenshaw-dev) [inline internal/controller/changetransferpolicy_controller.go:392]: Maybe collect a list and log all non-successful keys on one line?

**Pull request merged**